docs/src/pages/docs/network/01-custom-routing.mdx

```diff
@@ -0,0 +1,227 @@
+---
+title: "Custom Routing in Agent Networks | Network Documentation | Mastra"
+description: Learn how to implement custom routing logic in Mastra agent networks to control the flow between agents and create sophisticated multi-agent workflows.
+---
+
+# Custom Routing in Agent Networks
+
+Custom routing is a powerful feature of Agent Networks that allows you to control which agent is called next based on the current state, previous results, and other factors. This enables you to create sophisticated workflows where agents collaborate in a structured manner.
+
+## 1. Router Function Basics
+
+The router function determines which agent to call next in the network. It receives a `RouterState` object and returns either an agent or `undefined` (to signal completion):
+
+```ts showLineNumbers copy
+const network = new AgentNetwork({
+  name: "Custom Routing Example",
+  agents: [agentA, agentB, agentC],
+  router: async ({ callCount, state, lastResult, input }) => {
+    // Your routing logic here
+    if (callCount === 0) return agentA;
+    if (callCount === 1) return agentB;
+    return undefined; // Done after two calls
+  },
```

Comment:
Is the router fn optional? I was assuming the router model would choose which agent to call via tool calls or something like that, basically that a network would be a router agent with tools to access agent A,B,C as needed

Reply:
Ah I see that down below. Curious why we don't only go for that option to start - could reduce the API surface and complexity.
For use cases where you have agents and you want to go back and forth between them, that seems like it'd make more sense to just write a loop that calls them one after another. To me a network feels like it'd be most intuitive if the logic is exclusively decided by the router agent, otherwise I can just write code logic to select the next agent in a workflow or something

Reply:
I was going to ask the same question :joy: